src/main/java/org/galatea/starter/service/RestExceptionHandler.java

```diff
@@ -0,0 +1,51 @@
+package org.galatea.starter.service;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.galatea.starter.domain.exception.EntityNotFoundException;
+import org.springframework.dao.DataAccessException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+/**
+ * A centralized REST handler that intercepts exceptions thrown by controller calls, enabling a
+ * custom response to be returned.
+ *
+ * The returned ResponseEntity body object will be serialised into JSON (hence the need for the
+ * ApiError wrapper class).
+ */
+@ControllerAdvice
+@Slf4j
+public class RestExceptionHandler {
```

Comment:
This should be in the entrypoint package, given that it relates exclusively to rest services.